src/ui/youtube-modal/youtube-modal.ts

```diff
@@ -40,12 +40,12 @@ interface YouTubeModalConfig {
   /**
    * Callback triggered when the modal is opened.
    */
-  onModalOpen?: (ytModal: YouTubeModal) => void;
+  onModalOpen?: (ytModal: YouTubeModal, player: YT.Player) => void;
 
   /**
    * Callback triggered when the modal is closed.
    */
-  onModalClose?: (ytModal: YouTubeModal) => void;
+  onModalClose?: (ytModal: YouTubeModal, player: YT.Player) => void;
```

Comment:
Instead of this, add a `getPlayer()` or `getYTPlayer()` method to YouTubeModal, then you can call `ytModal.getPlayer()`.

Reply:
Done